src/map/script.cpp

```diff
@@ -21837,46 +21837,44 @@ BUILDIN_FUNC(instance_announce) {
  *------------------------------------------*/
 BUILDIN_FUNC(instance_check_party)
 {
-	int amount, min, max, i, party_id, c = 0;
-	struct party_data *p;
-
-	amount = script_hasdata(st,3) ? script_getnum(st,3) : 1; // Amount of needed Partymembers for the Instance.
-	min = script_hasdata(st,4) ? script_getnum(st,4) : 1; // Minimum Level needed to join the Instance.
-	max  = script_hasdata(st,5) ? script_getnum(st,5) : MAX_LEVEL; // Maxium Level allowed to join the Instance.
+	int32 min = script_hasdata(st, 4) ? script_getnum(st, 4) : 1; // Minimum Level needed to join the Instance.
+	int32 max  = script_hasdata(st, 5) ? script_getnum(st, 5) : MAX_LEVEL; // Maxium Level allowed to join the Instance.
 
-	if( min < 1 || min > MAX_LEVEL) {
+	if (min < 1 || min > MAX_LEVEL) {
 		ShowError("buildin_instance_check_party: Invalid min level, %d\n", min);
+		script_pushint(st, 0);
 		return SCRIPT_CMD_FAILURE;
-	} else if(  max < 1 || max > MAX_LEVEL) {
+	}
+	else if (max < 1 || max > MAX_LEVEL) {
 		ShowError("buildin_instance_check_party: Invalid max level, %d\n", max);
+		script_pushint(st, 0);
 		return SCRIPT_CMD_FAILURE;
 	}
 
-	if( script_hasdata(st,2) )
-		party_id = script_getnum(st,2);
-	else return SCRIPT_CMD_FAILURE;
+	int32 party_id = script_getnum(st, 2);
+	party_data *p = party_search( party_id );
 
-	if( !(p = party_search(party_id)) ) {
-		script_pushint(st, 0); // Returns false if party does not exist.
-		return SCRIPT_CMD_FAILURE;
+	if (p == nullptr) {
+		script_pushint(st, 0);
+		return SCRIPT_CMD_SUCCESS;
```

Comment:
Do we really support people like that? :-(
This is a command intended to check "A PARTY".

Reply:
We currently are using it like that :
https://github.com/rathena/rathena/blob/master/npc/instances/EndlessTower.txt#L213
Imo we could leave it like this, the command fails when there is no party and the command is always used as a condition. Also this would add a duplicate message in official instances.
```
	if (.@party < 1) {
		mes "Make or join a party with more than 1 member and try again.";
		close;
	}
	if (!instance_check_party(.@party_id,2)) {
		mes "Make or join a party with more than 1 member and try again.";
		close;
	}
```

Reply:
Ok you got me convinced. ;-)